extensions/typescript-language-features/src/languageFeatures/folding.ts

```diff
@@ -54,7 +59,7 @@ class TypeScriptFoldingProvider implements vscode.FoldingRangeProvider {
 		}
 
 		const start = range.start.line;
-		const end = this.adjustFoldingEnd(range, document);
+		const end = compact ? range.end.line : this.adjustFoldingEnd(range, document);
```

Comment:
Does compact folding need to do anything special for cases where the closing `}` also has text content on the same line? For example:

```ts
if (true) {
    console.log(1); }
```

and

```ts
if (true) {
      console.log(1); 
};   console.log(2); 
```